### DISCUSSION TITLE
Pre-compiled firmware files on github

### DISCUSSION THREAD

**Discussion by ksjh** [ksjh/silabs-firmware-builder#2] [Announcements]
Pre-compiled firmware files are available on github, but rather hard to find (github builds them automatically after commits to the repo). On the top of the github page, under "Actions", you can find previous runs and "Artifacts" (lower part of the page of one run). The names of the resulting files are not that clear, the file ``rcp-uart-802154-zbdonglee.zip`` is the right one for a ZBDongle-E with RTS-CTS handshaking with 115200 baud. Among other things, it contains the required firmware file ``build/release/rcp-uart-802154.gbl``. The naming convention is similar for other hardware. The ZB-GW04 v1.1 is the only firmware file that does **not** use hardware (RTS-CTS) handshaking.

Please make sure that you select appropriate options in the Silicon Labs Multiprotocol Home Assistant add-on.

**Comment by MichaelMaex** [comment]
Thank you very much! I flashed the rpc multipan firmware to my ZBDongle-E on Saturday and so far it's working great with my modest setup consisting of a handful thread and Zigbee devices.

**Comment by jusicgn** [comment]
I want to thank you too. I just flashed my ZBDongle-E with rcp-uart-802154-zbdonglee-none a few minutes ago. Had to manually disable the hardware flow control in the multiprotocol addon settings. Then reconfiguring the zha radio settings and uploading the backup worked like a charm. So far everything seems to work fine including OT. I'll now flash one of my eve devices to matter firmware to see if I can connect it using thread.

**Reply by ksjh** [reply]
Thank you, glad that it works. Is there any specific reason why you selected the version without hardware flow control? Versions with RTS/CTS also support higher baud rates, which might be beneficial in a scenario with both Zigbee and Thread/Matter devices.

**Reply by jusicgn** [reply]
Hi ksjh, not really a specific reason I just read in the other discussion there might be some problems with the hardware flow control. But maybe I've misunderstood that. As I can tell after almost 20h the zigbee devices are not that reliable/responsive as they were before. Sometimes they work sometimes they don't or need many seconds to respond. So I'm willing to try another firmware flavour. Do you recommend a specific one?

**Reply by ksjh** [reply]
Thank you for the feedback. My current test setup uses the ``rcp-uart-802154-zbdonglee-230`` variant. I did not notice any problems, so far. But it is just a small test network, not my complete home automation system. And it runs on an x86-64 thin client. This might make a difference, since with the RCP variants, Zigbee has to be handled by the CPU in software. So, a slow CPU might struggle a bit, depending on what else is happening on the system.

**Reply by MichaelMaex** [reply]
I am also using the -230 variant but on a rpi4. So far I haven't had any issues after the initial setup, but my setup is pretty small too. 
This might be a little of topic, but you could also use the eve devices with homekit over thread. Then there is no need to flash and that might be more stable than matter for now. I'm running my Nanoleaf lights with homekit over thread and they work great.

**Reply by jusicgn** [reply]
@ksjh thx for the reply. Just a few minutes before your reply I've flashed the `rcp-uart-802154-zbdonglee-460` variant. Flashing worked without any trouble. But I still have issues with the reliability/responsiveness of zigbee/zha. 
I'm running HAos on a Raspberry4 with 8GB RAM. Normal processor load is around 20%.
Reconfigured zha after flashing the new fw and uploaded the zha backup.
Before I go to try the `rcp-uart-802154-zbdonglee-230` variant: Should I try to reset zha to zero and start pairing devices from scratch or is it fine to go with the reconfigure/upload backup procedure?
I also have some log entries from zha/zigbee like:
`Protocol version 11 is not supported, using version 10 instead`
`ControllerApplication reset unsuccessful: ConnectionRefusedError(111, "Connect call failed ('172.30.32.1', 9999)")`

@MichaelMaex thx for the hint. At the moment I have one eve thermo and one eve energy and also one eve motion connected via homekitController. But that way the devices becoming unavailable randomly. Because of this I hope for better reliability by using matter/thread natively but want to have the zigbee part running stable before flashing the first eve devices with the matter fw.

**Reply by ksjh** [reply]
The processor should be powerful enough, the load is not too high. What looks very strange to me is the log entry about the protocol version. Is there an update available for zha? This seems not right, I have not seen this log entry before. But I also tend to use zigbee2mqtt, not zha (but tested zha in my test environment briefly). With the RCP firmware, zigbeed handles Zigbee in software. And both the Multiprotocol add-on (which includes zigbeed and thus specifies the protocol version used) and zha are provided by NabuCasa, so they are intended to work together well, whereas the zigbee2mqtt solution is more 3rd party and experimental.
I cannot really comment on the backup/restore. Technically, it should work without re-pairing. But I also read one other comment in this repo that the devices misbehaved (when switching from EmberZNet to RCPMultiPAN) until the devices were paired again.

**Reply by jusicgn** [reply]
@ksjh thx for the quick reply. I will try to delete zha und repair the devices from scratch then. Do you think I should do this with the actual `rcp-uart-802154-zbdonglee-460` or should I flash the `rcp-uart-802154-zbdonglee-230` variant before doing so?
Don't know if it's important but I grabbed the -460 variant from your last run https://github.com/ksjh/silabs-firmware-builder/actions/runs/4520545220.

**Reply by ksjh** [reply]
I think it should not matter too much, but I compiled the 460800 only yesterday, so this new variant has not been tested for long, whereas a number of persons have successfully used the 230400 baud version for a bit longer. I do not expect large problems with the 460800 baud variant, but have only little experience with it.

**Reply by jusicgn** [reply]
@ksjh alright, flashed the -230 variant and deleted the zha config. Then added a new zha config and repaired all devices. Seems to work for the moment even though the log still displays some of the errors/warnings. Will let that run a little and observe if it's more stable now and then report here.

**Reply by ksjh** [reply]
Are there still warnings/errors about the wrong protocol versions appearing?
When just the wireless communication is unreliable, perhaps [these tips](https://community.home-assistant.io/t/guide-for-zigbee-interference-avoidance-and-network-range-coverage-optimization/515752) might help to improve wireless reception. Of course, this does not help with the communication between the Pi and the dongle.

**Reply by jusicgn** [reply]
@ksjh yes, the warning about the protocol version is still there but seems to come up only once when starting zha:
```
Logger: bellows.ezsp
Source: components/zha/core/gateway.py:185
First occurred: 26. März 2023 um 21:05:25 (1 occurrences)
Last logged: 26. März 2023 um 21:05:25

Protocol version 11 is not supported, using version 10 instead
```
I also see the following error right before that warning
```
Logger: bellows.uart
Source: runner.py:179
First occurred: 26. März 2023 um 21:05:00 (2 occurrences)
Last logged: 26. März 2023 um 21:05:20

Lost serial connection: OSError('Server closed connection')
Received an unexpected reset: <NcpResetCode.RESET_SOFTWARE: 11>
```
Besides these everything seems to work after the reset of zha and repairing all devices.
Can't say much about the reliability, that needs some more time I think. But atm everything seems to work.
